src/googleCloud/DatasetSelector.js

```diff
@@ -0,0 +1,169 @@
+import DicomStorePicker from './DicomStorePicker';
+import DatasetPicker from './DatasetPicker';
+import ProjectPicker from './ProjectPicker';
+import LocationPicker from './LocationPicker';
+import './googleCloud.css';
+
+export default class DatasetSelector extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      project: null,
+      location: null,
+      dataset: null,
+      unloading: false,
+    };
+
+    this.onProjectSelect = this.onProjectSelect.bind(this);
+    this.onLocationSelect = this.onLocationSelect.bind(this);
+    this.onDatasetSelect = this.onDatasetSelect.bind(this);
+    this.onDicomStoreSelect = this.onDicomStoreSelect.bind(this);
+    this.onProjectClick = this.onProjectClick.bind(this);
+    this.onLocationClick = this.onLocationClick.bind(this);
+    this.onDatasetClick = this.onDatasetClick.bind(this);
+  }
+
+  static propTypes = {
+    id: PropTypes.string,
+    event: PropTypes.string,
+    oidcKey: PropTypes.string,
+    canClose: PropTypes.string,
+    setServers: PropTypes.func.isRequired,
+  };
+  static defaultProps = {};
+
+  onProjectSelect(project) {
+    this.setState({
+      project: project,
+    });
+  }
+
+  onLocationSelect(location) {
+    this.setState({
+      location: location,
+    });
+  }
+
+  onDatasetSelect(dataset) {
+    this.setState({
+      dataset: dataset,
+    });
+  }
+
+  onProjectClick() {
+    this.setState({
+      dataset: null,
+    });
+    this.setState({
+      location: null,
+    });
+    this.setState({
+      project: null,
+    });
+  }
+
+  onLocationClick() {
+    this.setState({
+      dataset: null,
+    });
+    this.setState({
+      location: null,
+    });
+  }
+
+  onDatasetClick() {
+    this.setState({
+      dataset: null,
+    });
+  }
+
+  onDicomStoreSelect(dicomStoreJson) {
+    const dicomStore = dicomStoreJson.name;
+    const parts = dicomStore.split('/');
+    const result = {
+      wadoUriRoot: `https://healthcare.googleapis.com/v1beta1/${dicomStore}/dicomWeb`,
+      qidoRoot: `https://healthcare.googleapis.com/v1beta1/${dicomStore}/dicomWeb`,
+      wadoRoot: `https://healthcare.googleapis.com/v1beta1/${dicomStore}/dicomWeb`,
+      project: parts[1],
+      location: parts[3],
+      dataset: parts[5],
+      dicomStore: parts[7],
+    };
+    this.props.setServers(result);
+  }
+
+  render() {
+    const { project, location, dataset } = this.state;
+    const {
+      onProjectClick,
+      onLocationClick,
+      onDatasetClick,
+      onProjectSelect,
+      onLocationSelect,
+      onDatasetSelect,
+      onDicomStoreSelect,
+    } = this;
+    return (
+      <>
+        <span className="gcp-picker--title">Google Cloud Healthcare API</span>
+        {project && (
+          <div className="gcp-picker--path">
+            <span onClick={() => onProjectClick(null)}>{project.name}</span>
+            {project && location && (
+              <>
+                <span onClick={() => onLocationClick(null)}>
```

Comment:
Can't you do this?
`onClick={onLocationClick}`

Reply:
done